### PR TITLE
TimeSeries: Add data frame index and field name

### PR DESCRIPTION
**Pull request by dprokop**
Ref #18918 

Adds data frame view and field name to the TimeSeries for later inspection, in graph panel data links in particular

### CODE REVIEW

**Comment by torkelo** [comment]
Is there no other way? adding it to this legacy object could be a bit problematic, we send it in pure to flot when it renders the graph and flot does some deep cloning. So would prefer to not DataFrameView on it.

**Comment by dprokop** [comment]
Other way is to simply add index of the original dataframe and retrieve it that way

File: public/app/core/time_series2.ts [hunk 1]
```diff
@@ -70,6 +70,8 @@ export function getDataMinMax(data: TimeSeries[]) {
 export default class TimeSeries {
   datapoints: any;
   id: string;
+  dataFrameIndex: number;
+  fieldName: string;
```
Comment (ryantxu) [inline public/app/core/time_series2.ts]: Can we add some comments to these fields that say why they are needed and maybe a comment saying this whole class is deprecated and should not be used in any new panels?

File: public/app/plugins/panel/graph/data_processor.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export class DataProcessor {
       return list;
     }
 
-    for (const series of dataList) {
+    for (const [index, series] of dataList.entries()) {
```
Comment (ryantxu) [inline public/app/plugins/panel/graph/data_processor.ts]: minor nit -- that I know @torkelo agrees with...  what about just a regular `for(let i=0,...` loop?

**Review by ryantxu — APPROVED** [review]
Add some comments and drop the `.entries()` call, then LGTM

File: public/app/plugins/panel/graph/data_processor.ts (hunk 2 above)
Comment (torkelo) [reply public/app/plugins/panel/graph/data_processor.ts]: yea, commented on this one in the other PR (this is part of the other PR as well). entries() seems like a scary function to use for loops as it creates a whole new array with index,value pairs.

File: public/app/core/time_series2.ts [hunk 3]
```diff
@@ -67,10 +67,20 @@ export function getDataMinMax(data: TimeSeries[]) {
...
 export default class TimeSeries {
   datapoints: any;
   id: string;
+  // Represents index of original data frame in the quey response
   dataFrameIndex: number;
+  /**
+   * Name of the field the time series was created from
+   * Used in graph panel to retrieve value from the original data frame
+   */
   fieldName: string;
```
Comment (torkelo) [inline public/app/core/time_series2.ts]: Any reason for not using fieldIndex? Field names are not enforced to be unique so will be more robust to use fieldIndex to loop up a field later.

Comment (dprokop) [reply public/app/core/time_series2.ts]: Good point. Missed removing this as we already have field index available in the TimeSeries

**Pull request merged**